cloud.ganttproject.colloboque/src/main/kotlin/cloud/ganttproject/colloboque/PostgreStorageApi.kt

```diff
@@ -112,6 +117,60 @@ class PostgreStorageApi(private val factory: ConnectionFactory, private val supe
     }
   }
 
+  fun parallelTransactions(
+    projectXml: String,
+    baseTxnId: BaseTxnId,
+    serverTransaction: List<XlogRecord>,
+    clientTransaction: List<XlogRecord>
+  ): Boolean {
+    val settings = Settings().withRenderNameCase(RenderNameCase.LOWER)
+      .withInterpreterDialect(SQLDialect.POSTGRES)
+    val jdbcUrl = "jdbc:postgresql://localhost:5432/${baseTxnId}_${Random.nextLong()};DB_CLOSE_DELAY=-1;DATABASE_TO_LOWER=true"
```

Comment:
Let's use our `factory: ConnectionFactory` instance to create connections. It probably makes sense to replace `DataSource` in `SqlProjectDatabaseImpl` constructor with `ConnectionFactory` because it needs only a connection anyway.

Reply:
done